modules/spring-config/src/main/java/org/mule/config/spring/parsers/specific/TransformerMessageProcessorDefinitionParser.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.config.spring.parsers.specific;
+
+import org.mule.transformer.types.SimpleDataType;
+
+import org.springframework.beans.MutablePropertyValues;
+import org.springframework.beans.PropertyValues;
+import org.springframework.beans.factory.support.AbstractBeanDefinition;
+import org.springframework.beans.factory.support.BeanDefinitionBuilder;
+import org.springframework.beans.factory.xml.ParserContext;
+import org.w3c.dom.Element;
+
+public class TransformerMessageProcessorDefinitionParser extends MessageProcessorDefinitionParser
+{
+
+    public static final String RETURN_CLASS = "returnClass";
+    public static final String ENCODING = "encoding";
+    public static final String MIME_TYPE = "mimeType";
+
+    public TransformerMessageProcessorDefinitionParser(Class messageProcessor)
+    {
+        super(messageProcessor);
+    }
+
+    public TransformerMessageProcessorDefinitionParser()
+    {
+        super();
+    }
+
+    @Override
+    protected AbstractBeanDefinition parseInternal(Element element, ParserContext parserContext)
+    {
+        final AbstractBeanDefinition abstractBeanDefinition = super.parseInternal(element, parserContext);
+
+        MutablePropertyValues props = abstractBeanDefinition.getPropertyValues();
+        final AbstractBeanDefinition dataTypeBeanDefinition = parseDataType(props);
+        if (dataTypeBeanDefinition != null)
+        {
+            props.add("returnDataType", dataTypeBeanDefinition);
+            props.removePropertyValue(RETURN_CLASS);
```

Comment:
add javadoc explaining why these properties are removed.. I don't get it at a first glance